Ladybird/Utilities.cpp

```diff
@@ -12,45 +12,61 @@
 #include <LibCore/System.h>
 #include <LibFileSystem/FileSystem.h>
 
+#define TOKENCAT(x, y) x##y
+#define STRINGIFY(x) TOKENCAT(x, sv)
+
+// This is expected to be set from the build scripts, if a packager desires
+#if defined(LADYBIRD_LIBEXECDIR)
+constexpr auto libexec_path = STRINGIFY(LADYBIRD_LIBEXECDIR);
+#else
+constexpr auto libexec_path = "libexec"sv;
+#endif
+
 ByteString s_serenity_resource_root;
 
-ErrorOr<String> application_directory()
+ErrorOr<ByteString> application_directory()
 {
     auto current_executable_path = TRY(Core::System::current_executable_path());
-    auto dirname = LexicalPath::dirname(current_executable_path);
-    return String::from_byte_string(dirname);
+    return LexicalPath::dirname(current_executable_path);
 }
 
 void platform_init()
 {
     s_serenity_resource_root = [] {
-        auto const* source_dir = getenv("SERENITY_SOURCE_DIR");
-        if (source_dir) {
-            return ByteString::formatted("{}/Base", source_dir);
-        }
         auto* home = getenv("XDG_CONFIG_HOME") ?: getenv("HOME");
-        VERIFY(home);
-        auto home_lagom = ByteString::formatted("{}/.lagom", home);
-        if (FileSystem::is_directory(home_lagom))
-            return home_lagom;
-        auto app_dir = application_directory().release_value_but_fixme_should_propagate_errors().to_byte_string();
+        if (home != nullptr) {
```

Comment:
I was told by the FreeBSD packager that some configs don't have either XDG_CONFIG_HOME or HOME? wild. Anyway, let's make them not have to patch....

Reply:
`HOME` is usually a given, but you cannot assume that any of the `XDG_...` variables are set.  Hence why `StandardPaths::runtime_directory()` failed on FreeBSD before a fallback was added.

Reply:
Gotcha, I was misremembering then. I'll keep this change though, it probably doesn't hurt.